src/App.tsx

```diff
@@ -13,38 +14,117 @@ export const goodsFromServer = [
   'Honey',
   'Jam',
   'Garlic',
-];
+] as const;
+
+enum SortStatus {
+  alphabetically = 'alphabetically',
+  length = 'length',
+  default = '',
+}
+
+const getUpdatedList = (
+  goods: readonly string[],
+  statusOfCurrentList: SortStatus,
+  isReversed: boolean,
+): string[] => {
+  let preparedGoods = [...goods];
+
+  if (statusOfCurrentList) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (statusOfCurrentList) {
+        case SortStatus.alphabetically:
+          return good1.localeCompare(good2);
+        case SortStatus.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
 
 export const App: React.FC = () => {
+  const [statusOfCurrentList, setStatusOfCurrentList] = useState(
+    SortStatus.default,
+  );
+  const [isReversed, setIsReversed] = useState(false);
+
+  const showReset = statusOfCurrentList !== SortStatus.default || isReversed;
+  const visibleGood = getUpdatedList(
+    goodsFromServer,
+    statusOfCurrentList,
+    isReversed,
+  );
+  const reset = () => {
+    setStatusOfCurrentList(SortStatus.default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames({
+            'button is-info': true,
+            'is-light': statusOfCurrentList !== SortStatus.alphabetically,
```

Comment:
```suggestion
          className={classNames( 'button is-info', { 'is-light': statusOfCurrentList !== SortStatus.alphabetically,
```

You can add classes without a condition